funcframework/framework.go

```diff
@@ -141,6 +141,9 @@ func Start(port string) error {
 			handler = server
 		}
 	}
+	if handler == nil {
+		return fmt.Errorf("no matching function in the registry")
```

Comment:
The registry is mostly an internal concept to the Functions Framework. The API users see is just `functions.HTTP`, so they aren't aware of the registry, so we probably want to be more generic with something like "no function matching <insert intended FUNCTION_TARGET> found"